### PR TITLE
FIX Replaced deprecated React functions

### PR DESCRIPTION
**Pull request by sabina-talipova**
### Description
- Replaced deprecated React function `componentWillReceiveProps` with `componentDidUpdate`

### Parent issue
- https://github.com/silverstripe/silverstripe-admin/issues/1416

#### Note
There are a few warnings that should be fixed in another modules such as `silverstripe/admin`, etc.
 - `Warning: Injector.getDerivedStateFromProps(): A valid state object (or null) must be returned. You have returned undefined.` happens in `silverstripe\admin`
 - `Deprecation Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.` exists in `silverstripe/admin/client/src/styles/_variables.scss`
 
 These issues will be fixed be PR

### CODE REVIEW

File: client/src/containers/Gallery/Gallery.scss [hunk 1]
```diff
@@ -225,7 +225,7 @@
     background-image: none;
     text-align: center;
     display: inline;
-    @include form-control-focus();
+    @include form-control-focus(true);
```
Comment (sabina-talipova) [inline client/src/containers/Gallery/Gallery.scss]: `form-control-focus` is deprecated https://github.com/twbs/bootstrap/issues/29096.
I just suppress a warning, but probably we could remove this include.

File: client/src/components/AssetDropzone/AssetDropzone.js [hunk 2]
```diff
@@ -52,9 +52,15 @@ class AssetDropzone extends Component {
     }
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentDidUpdate(prevProps) {
```
Comment (maxime-rainville) [inline client/src/components/AssetDropzone/AssetDropzone.js:55]: I'm going to assume this is related to this. When I upload multiple files all at once with this PR enabled, I get weird behaviour where the files don't appear to complete.

[Screencast from 2023-03-21 14-11-16.webm](https://user-images.githubusercontent.com/1168676/226497790-6503c9a0-a95a-4bc8-9503-70bdb2b880e8.webm)

In this video, I start out without the PR and upload a bunch of files successfully. I swap to the PR ... some files are uploaded successfully but not some others. I swap back to the 2.0 branch, and upload work again.

File: client/src/components/GalleryItem/GalleryItem.js [hunk 3]
```diff
@@ -49,12 +49,12 @@ class GalleryItem extends Component {
     this.handleCancelUpload = this.handleCancelUpload.bind(this);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (shouldLoadImage(nextProps)) {
+  componentDidUpdate() {
+    if (shouldLoadImage(this.props)) {
```
Comment (maxime-rainville) [inline client/src/components/GalleryItem/GalleryItem.js:53]: This seems to be working fine.

File: client/src/components/InsertEmbedModal/InsertEmbedModal.js [hunk 4]
```diff
@@ -16,13 +16,13 @@ class InsertEmbedModal extends Component {
...
+  componentDidMount() {
     this.setOverrides(this.props);
   }
 
-  componentWillReceiveProps(props) {
-    if (props.isOpen && !this.props.isOpen) {
-      this.setOverrides(props);
+  componentDidUpdate(prevProps) {
+    if (this.props.isOpen && !prevProps.isOpen) {
+      this.setOverrides(this.props);
```
Comment (maxime-rainville) [inline client/src/components/InsertEmbedModal/InsertEmbedModal.js:25]: This still seems to work fine.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Just partial review.

File: client/src/containers/AssetAdmin/AssetAdmin.js [hunk 5]
```diff
@@ -54,9 +54,9 @@ class AssetAdmin extends Component {
     this.handleMoveFilesSuccess = this.handleMoveFilesSuccess.bind(this);
   }
 
-  componentWillReceiveProps(props) {
-    if ((typeof props.onReplaceUrl === 'function') && !props.loading && props.folder && props.folderId !== props.folder.id) {
-      props.onReplaceUrl(props.folder.id, props.fileId, props.query, props.viewAction);
+  componentDidUpdate() {
```
Comment (maxime-rainville) [inline client/src/containers/AssetAdmin/AssetAdmin.js:57]: Seems to be working fine.

File: client/src/containers/Gallery/Gallery.js [hunk 6]
```diff
@@ -77,29 +77,30 @@ class Gallery extends Component {
     window.addEventListener('keyup', this.toggleSelectConcat);
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentDidUpdate(prevProps) {
+    this.initSortDropdown();
+    this.initFlushUploadFiles(prevProps);
+  }
+
+  componentWillUnmount() {
```
Comment (maxime-rainville) [inline client/src/containers/Gallery/Gallery.js:85]: All of this still seems to work fine.

File: client/src/containers/Gallery/Gallery.scss (hunk 1 above)
Comment (maxime-rainville) [reply client/src/containers/Gallery/Gallery.scss]: Looking at the [matching PR on bootstrap](https://github.com/twbs/bootstrap/pull/28262/files) it looks like we could simply copy past this snippet here instead.

```scss
&:focus {
    color: $input-focus-color;
    background-color: $input-focus-bg;
    border-color: $input-focus-border-color;
    outline: 0;
    // Avoid using mixin so we can pass custom focus shadow properly
    @if $enable-shadows {
      box-shadow: $input-box-shadow, $input-focus-box-shadow;
    } @else {
      box-shadow: $input-focus-box-shadow;
    }
  }
```

File: client/src/containers/Gallery/GalleryDND.js [hunk 7]
```diff
@@ -23,7 +23,7 @@ class GalleryDND extends Component {
     window.addEventListener('drop', this.handleDrop, true);
   }
 
-  componentWillUpdate() {
+  componentDidUpdate() {
```
Comment (maxime-rainville) [inline client/src/containers/Gallery/GalleryDND.js:26]: Works fine.

File: client/src/containers/HistoryList/HistoryList.js [hunk 8]
```diff
@@ -47,9 +47,9 @@ class HistoryList extends Component {
     this.refreshHistoryIfNeeded();
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentDidUpdate(prevProps) {
```
Comment (maxime-rainville) [inline client/src/containers/HistoryList/HistoryList.js:50]: Didn't even know this thing existed. I've enabled it and it seems to be working.

File: client/src/containers/InsertMediaModal/InsertMediaModal.js [hunk 9]
```diff
@@ -37,17 +37,17 @@ class InsertMediaModal extends Component {
     }
   }
 
-  componentWillReceiveProps(props) {
-    if (!props.isOpen && this.props.isOpen) {
-      props.onBrowse(props.folderId);
-      props.actions.gallery.deselectFiles();
+  componentDidUpdate(prevProps) {
```
Comment (maxime-rainville) [inline client/src/containers/InsertMediaModal/InsertMediaModal.js:40]: Still seems to be working fine.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Completed review. Just need to fix the issue when uploading multiple files. 

I think the proper fix for the deprecated `form-control-focus`  is to include the appriate rules directly in the component.

File: client/src/containers/Gallery/Gallery.scss (hunk 1 above)
Comment (sabina-talipova) [reply client/src/containers/Gallery/Gallery.scss]: DONE

File: client/src/components/AssetDropzone/AssetDropzone.js (hunk 2 above)
Comment (sabina-talipova) [reply client/src/components/AssetDropzone/AssetDropzone.js:55]: I've fixed it. Problem was in another file.

Comment (maxime-rainville) [reply client/src/components/AssetDropzone/AssetDropzone.js:55]: Looks like it works properly now. Can you point me to what you actually did to fix it?

**Review by maxime-rainville — APPROVED** [review]

**Pull request merged**